### ISSUE TITLE
Display stays on after screen activation e.g. due push notifification

### ISSUE THREAD

**Issue by schnedann** [osmandapp/OsmAnd#19056]
### Description

This is an old issue, existing since I use OSMAND. But it also exits with my new device on modern Android () and latest OSMAND from FDroiod.

When arrived at some place one might lay the device somewhere (floor, desk,...). OSMAND was the last App used (e.g. Car navigation). Routing is stopped (no active route). When some App (e.g weather information, chat,...) receives a push notification the screen is activated, OSMAND shows up and the screen stays on as long the battery lasts.

### Steps to reproduce

* do some navigation
* reach endpoint, stop navigation --> no active route
* lay down device in house
* wait for a push notification

### Actual result

* screen is turned on and stays active and drains battery (+ possible burn in of oled screen!)

### Expected result

* screen is turned on due push notification
* OSMAND detects no movement at all (sensors, gps,...) and allows the screen to turn off again
* push OSMAND to background, so its not the first app which gets activated on next push notification

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version:
Android/iOS version:
Device model:
Crash-Logs: ?
```

**Comment by sonora** [comment]
Yes, now that you raise this issue it also fits to some complaints which had reached me over the yesrs where users observed the following:

They are in their car, phone in their pocket but connected to the car's hands-free speaker phone function. While driving, they receive an incoming call which they accept using the button on their steering wheel. It seems that after the call has ended OsmAnd may have somehow become active, because some time later their battery is drained and it appears OsmAnd is the culprit.

**Comment by schnedann** [comment]
sounds similar enough to be the same problem.

I think at least if there is no active route and / or no movement for the last say 5 minutes an app like OSMAND can detect it should not be active in foreground endlessly. Might be impossible if the device is carried / moved some way, so here at least one can evaluate if there is an active route / navigation...

but an improved behaviour should be possible

**Comment by sjvudp** [comment]
Meta: Original author, please correct the "fifi" in the title ("notification"); I cannot edit ;-)

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.9.0#2932m, released: 2024-08-07
In my case, the issue can be reproduced only in real life conditions when:
- you have followed navigation instructions and arrived at your destination
- if there is no more movement, the screen of my phone turns off automatically
- but when you leave your car at the parking slot and go where you need to go, the screen turns on as if you continue to move on your car. 
- if you go far away, then you even may hear notification about route recalculation

It's difficult to film that behavior.

**Comment by schnedann** [comment]
that sounds, you still have an active route. Which is also an "issue" as one might want to pause routing e.g. when you stop on a shop on your way. currently - as far as I know - you need to stop / delete the active route and restart it later.

**Comment by sonora** [comment]
There's a Pause button in Android's system notification for the navigation.

**Comment by schnedann** [comment]
Ah, yes, I read this before,... but why its not accessible via the normal GUI? I (my brain) normally ignores the system notification area - thats not what feels natural to operate an program.
In fact: when I will use OSMAND again next week, I will have forgotten about it already... at least it was the case last time.

**Comment by DmitryAlexei** [comment]
Issue was also reproduced without starting navigation.
- I've been testing the compass behavior at night and switched off the phone when OsmAnd was on the screen
- in the morning, the phone was activated by the alarm at 09:30 and 09:40 (both times I switched them off using my watch)
- at 09:40 after turning alarm off, phone's screen blinked several times and OsmAnd was shown
- but after tapping the menu button in the OsmAnd app, the phone has locked itself. 

Here is the logcat file 

[logcat (2).log.txt](https://github.com/user-attachments/files/16904714/logcat.2.log.txt)
